sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -498,21 +473,15 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
       throw new TypeError("The parameter 'onError' must be of type 'function'.");
     }
 
-    this._createMessageSessionIfDoesntExist()
-      .then(async () => {
-        if (!this._messageSession) {
-          return;
-        }
-        if (!this._isClosed) {
-          this._messageSession.receive(onMessage, onError, options);
-        } else {
-          await this._messageSession.close();
-        }
-        return;
-      })
-      .catch((err) => {
+    if (!this._isClosed) {
+      try {
+        this._messageSession.receive(onMessage, onError, options);
+      } catch (err) {
         onError(err);
-      });
+      }
+    } else {
+      this._messageSession.close().catch((err) => onError(err));
```

Comment:
Is `this._isClosed` is set do we need to close the session here? Wouldn't that imply that it's already being closed?

Reply:
Previously, we had this check to catch the case where user closes the receiver when `_createMessageSessionIfDoesntExist` had not yet completed.

Now that we are passing the messageSession to the SessionReceiverImpl, we do not need this.
Good catch!

Reply:
Sounds reasonable, this need not be here.

Reply:
Removed else.